src/scores/probability/crps_impl.py

```diff
@@ -869,6 +872,15 @@ def crps_for_ensemble(
     fcst_obs_term = abs(fcst - obs).mean(dim=ensemble_member_dim)
     result = fcst_obs_term - fcst_spread_term
 
+    if decomposition:
+        mask = ~np.isnan(fcst) * ~np.isnan(obs)  # create mask so that we can preserve NaNs
```

Comment:
I don't think it's significant, but it might be faster to use logical_and instead of performing a multiplication. In a lower-level language it would be faster but there is a lot of optimisation in the library and so I wouldn't assume in this case. Any speed difference may be undetectably small in this context. But the thought occurred to me so I thought I would mention it.

Reply:
Updated. Even if it's not meaningfully faster, your suggestion makes it clearer to read.